### DISCUSSION TITLE
System ordering not working

### DISCUSSION THREAD

**Discussion by fluffysquirrels** [bevyengine/bevy#3174] [Q&A] [answered]
I'm trying to run two systems one after the other, with the first system inserting a resource and the second system using the resource. This is failing with error: `Requested resource does not exist: my_project::enemy::Sprites`

Here's my code:
```rust
struct Sprites {
    idle: Handle<ColorMaterial>,
}

impl bevy::app::Plugin for Plugin {
    fn build(&self, app: &mut AppBuilder) {
        app
            .add_startup_system(load_resources.system().label("load"))
            .add_startup_system(spawn_current_enemy.system().after("load"));
    }
}

fn load_resources(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<ColorMaterial>>,
) {
    commands.insert_resource(Sprites {
        idle: materials.add(asset_server.load("x.png").into()),
    });
}

fn spawn_current_enemy(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<ColorMaterial>>,
    sprites: Res<Sprites>,
) {
    commands.spawn_bundle(SpriteBundle {
        material: sprites.idle.clone(),
        .. Default::default()
    });
}
```

Any ideas how to implement this correctly?

**Comment by alice-i-cecile** [comment, marked as answer]
Commands are not processed until the end of the current stage, so while the systems are running in the correct order the resource won't exist in time.

You need to split which stages these systems are in, by using `add_system_to_stage` and selecting one of the variants of the [StartupStage enum](https://docs.rs/bevy/0.5.0/bevy/app/enum.StartupStage.html#variant.Startup).

**Reply by alice-i-cecile** [reply]
In 0.10, commands are only processed after `apply_system_buffers`, which is a special system. By default, there's a copy of this system inserted between base system sets. You can add more though, and either use `(system_a, apply_system_buffers, system_b).chain()` or just give a unique label to your copy of `apply_system_buffers`.

In 0.11, base system sets are no more, and this simply uses schedules. `apply_system_buffers` has been renamed to `apply_deferred`, and deferred system parameters (like Commands) are automatically applied at the end of each schedule. You can still insert your own copies of `apply_deferred` within a schedule if you'd like as well!

Note that inserting exclusive systems like apply_system_buffers / apply_deferred will create a choke point for your parallel system evaluation: all running systems must complete before the schedule can advance. Not a big deal, but something to be mindful of at times.

**Comment by MinerSebas** [comment]
`.label()` + `.before()`/`.after()` is only for ordering inside a Stage, but Commands are only applied at the end of a stage.

Solution:
```rs
		app
            .add_startup_system_to_stage(StartupStage::PreStartup, load_resources.system())
            .add_startup_system(spawn_current_enemy.system());
```

**Reply by alice-i-cecile** [reply]
0.10: `app.add_startup_system(load_resources.in_base_set(StartupStage::PreStartup)` (now a base set)

0.11: `app.add_systems(PreStartup, load_resources)` (just a schedule)

**Comment by Ewan88** [comment]
Can these answers please be updated to v.0.10?

**Reply by alice-i-cecile** [reply]
Sure, I'll follow up below my answer.
